src/facet/inspection/_shap_global_explanation.py

```diff
@@ -88,27 +110,66 @@ def __init__(self, affinity_rel_ij: np.ndarray, std_p_i: np.ndarray) -> None:
 
         # relative symmetrical affinity is absolute symmetrical affinity scaled back
         # from total feature importance per feature pair
-        affinity_rel_sym_ij = affinity_abs_sym_ij_2x / (
-            importance_ij + transpose(importance_ij)
+        affinity_rel_sym_ij = np.zeros(affinity_rel_ij.shape)
+        np.divide(
+            affinity_abs_sym_ij_2x,
+            importance_ij + transpose(importance_ij),
+            out=affinity_rel_sym_ij,
+            # do not divide where the nominator is 0 (the denominator will be 0 as well)
+            where=affinity_abs_sym_ij_2x > 0.0,
```

Comment:
`affinity_abs_sym_ij_2x > 0.0` 
it could be worth a check if all zero-values are indeed represented by `0.0`, or, if calculated prior using lossly floating point arithmetics, are actually represented by something like `0.0000000001` (or similar). The division error is circumvented by the predicate in any case, however rounding to zero here could lead to more consistent results (treating all 'true' zeroes equally).

Reply:
We just want to prevent the case where we divide 0.0 by 0.0. If importance is 0, so is affinity (see multiplication in lines 106–109). If affinity is a very small value, we don't really care: the denominator will be in a similar order of magnitude given lines 106–109.

Reply:
I'm fine with that 👍  Thanks for checking